kubespawner/spawner.py

```diff
@@ -2455,31 +2463,64 @@ def _start_watching_events(self, replace=False):
         If replace=True, a running pod reflector will be stopped
         and a new one started (for recovering from possible errors).
         """
-        return self._start_reflector(
+        return await self._start_reflector(
             kind="events",
             reflector_class=EventReflector,
             fields={"involvedObject.kind": "Pod"},
             omit_namespace=self.enable_user_namespaces,
             replace=replace,
         )
 
-    def _start_watching_pods(self, replace=False):
-        """Start the pod reflector
+    async def _start_watching_pods(self, replace=False):
+        """Start the pods reflector
 
         If replace=False and the pod reflector is already running,
         do nothing.
 
         If replace=True, a running pod reflector will be stopped
         and a new one started (for recovering from possible errors).
         """
-        return self._start_reflector(
+        return await self._start_reflector(
             kind="pods",
             reflector_class=PodReflector,
             labels={"component": self.component_label},
             omit_namespace=self.enable_user_namespaces,
             replace=replace,
         )
 
+    async def _stop_reflector(self, kind: str):
+        """Stop a shared reflector on the KubeSpawner class"""
+
+        key = self._get_reflector_key(kind)
+        reflector = self.__class__.reflectors.pop(key, None)
+
+        if reflector:
+            await reflector.stop()
+
+    async def _stop_watching_pods(self):
+        """Stop pods reflector"""
+        await self._stop_reflector("pods")
+
+    async def _stop_watching_events(self):
+        """Stop events reflector"""
+        await self._stop_reflector("events")
+
+    async def _stop_watchers(self):
+        """Stop reflectors for current spawner instance"""
+        await self._stop_reflector()
+        if self.events_enabled:
+            await self._stop_watching_events()
+
+    @classmethod
+    async def _stop_all_reflectors(cls):
+        """Stop reflectors for all instances"""
```

Comment:
I saw that this was soleley used for tests, maybe it belongs in the conftest.py, but it could live here as well - I figure it merits a clarification though.

```suggestion
        """Stop reflectors for all instances, a function used when running tests."""
```

Reply:
Done